src/osp/Active/SysNewton.h

```diff
@@ -193,77 +197,51 @@ class SysNewton : public IDynamicSystem
     static void create_body(ActiveScene& rScene, ActiveEnt entity,
                             NewtonWorld const* nwtWorld);
 
+    /**
+     * Update the inertia properties of a rigid body
+     *
+     * Given an existing rigid body, computes and updates the mass matrix and
+     * center of mass. Entirely self contained, calls the other inertia
+     * functions in this class.
+     * @param entity [in] The rigid body to update
+     */
+    static void compute_rigidbody_inertia(ActiveScene& rScene, ActiveEnt entity);
+
     /**
     * Recursively compute the center of mass of a hierarchy subtree
     * 
-    * Takes in a root entity and recurses through its children. Entities which
-    * possess an ACompMass component are used to compute a center of mass for
-    * the entire subtree, treating it as a system of point masses.
+    * Takes in a root entity and recurses through its children. Entities which 
+    * possess an ACompMass component are used to compute a center of mass for 
+    * the entire subtree, treating it as a system of point masses. By default,
+    * the root entity's mass is not included; to include it, the optional
+    * includeRootMass argument can be set to 'true'.
     * 
     * @param rScene           [in] ActiveScene containing relevant scene data
     * @param root             [in] Entity at the root of the hierarchy subtree
-    * @param currentTransform [in] Transformation of root with respect to the
-    *                              initial call's root
+    * @param includeRootMass  [in] Set to true if the root entity's mass should
```

Comment:
should this really be a runtime option? or a compile time option?

You could make this a non-type template parameter (bool), and then use if constexpr inside the function to omit the extra code at runtime for code that doesn't want the mass added.

Reply:
I considered that, but the interface is a little ugly. `compute_hier_CoM<true>()` isn't the most intuitive thing; I guess in either case you still have to check the header to see what the function actually does though, so maybe it's worth it.

Reply:
Use an enum as the template type.

compute_hier_CoM<ERootMass::Include> vs compute_hier_CoM<ERootMass::Ignore>

Reply:
Shame there is no `switch constexpr(val)` syntax.